src/array.cr

```diff
@@ -1770,4 +1770,12 @@ class Array(T)
       end
     end
   end
+
+  def bsearch
+    idx=(0...self.size).bsearch{|i|yield self[i]}
+    idx ? self[idx] : nil
```

Comment:
Do you mean `raise unless idx`? Well, it might be possible since although we usually do `ary.bsearch_index{}||ary.size`, Array#bsearch is not the case.